bigtable/client/instance_admin_test.cc

```diff
@@ -488,8 +488,278 @@ TEST_F(InstanceAdminTest, CreateInstancePollReturnsFailure) {
+                          btproto::PartialUpdateInstanceRequest const& request,
+                          google::longrunning::Operation* response) {
+        auto const instance_name =
+            "projects/my-project/instances/test-instance";
+        EXPECT_EQ(instance_name, request.instance().name());
+        return grpc::Status::OK;
+      }));
+
+  std::string expected_text = R"(
+name: 'projects/my-project/instances/test-instance'
+display_name: 'foo bar'
+state: READY
+type: PRODUCTION
+labels: {
+  key: 'foo1'
+  value: 'bar1'
+}
+labels: {
+  key: 'foo2'
+  value: 'bar2'
+}
+)";
+  btproto::Instance expected;
+  ASSERT_TRUE(
+      google::protobuf::TextFormat::ParseFromString(expected_text, &expected));
+
+  std::string update_mask_text = R"(
+  paths: 'display_name'
+  paths: 'labels'
+  )";
+  google::protobuf::FieldMask update_mask;
+  ASSERT_TRUE(google::protobuf::TextFormat::ParseFromString(update_mask_text,
+                                                            &update_mask));
+
+  EXPECT_CALL(*client_, GetOperation(_, _, _))
+      .WillOnce(Invoke([](grpc::ClientContext*,
+                          google::longrunning::GetOperationRequest const&,
+                          google::longrunning::Operation* operation) {
+        operation->set_done(false);
+        return grpc::Status::OK;
+      }))
+      .WillOnce(Invoke([](grpc::ClientContext*,
+                          google::longrunning::GetOperationRequest const&,
+                          google::longrunning::Operation* operation) {
+        operation->set_done(false);
+        return grpc::Status::OK;
+      }))
+      .WillOnce(Invoke(
+          [&expected](grpc::ClientContext*,
+                      google::longrunning::GetOperationRequest const& request,
+                      google::longrunning::Operation* operation) {
+            operation->set_done(true);
+            auto any = bigtable::internal::make_unique<google::protobuf::Any>();
+            any->PackFrom(expected);
+            operation->set_allocated_response(any.release());
+            return grpc::Status::OK;
+          }));
+
+  auto future = tested.UpdateInstance(&expected, &update_mask);
+  auto actual = future.get();
+
+  std::string delta;
+  google::protobuf::util::MessageDifferencer differencer;
+  differencer.ReportDifferencesToString(&delta);
+  EXPECT_TRUE(differencer.Compare(expected, actual)) << delta;
+}
+
+/// @test Verify that `bigtable::InstanceAdmin::UpdateInstance` works.
+TEST_F(InstanceAdminTest, UpdateInstanceImmediatelyReady) {
+  using ::testing::_;
+  using ::testing::Invoke;
+
+  bigtable::InstanceAdmin tested(client_);
+
+  std::string expected_text = R"(
+name: 'projects/my-project/instances/test-instance'
+display_name: 'foo bar'
+state: READY
+type: PRODUCTION
+)";
+  btproto::Instance expected;
+  ASSERT_TRUE(
+      google::protobuf::TextFormat::ParseFromString(expected_text, &expected));
+  std::string update_mask_text = R"(
+paths: 'display_name'
```

Comment:
See other comment about indentation for these strings.

Reply:
fixed